pages/contest/[id].js

```diff
@@ -0,0 +1,147 @@
+  const [submissionDescription, setSubmissionDescription] = useState("");
+
+  // Uploads photo to show before submitting to server
+  function uploadToClient(event) {
+    if (event.target.files && event.target.files[0]) {
+      const i = event.target.files[0];
+      const reader = new FileReader();
+
+      reader.onloadend = () => {
+        setCreateObjectURL(reader.result);
+      };
+
+      reader.readAsDataURL(i);
+    }
+  }
+
+  function changeDescription(event) {
+    setSubmissionDescription(event.target.value);
+  }
+
+  async function uploadToServer() {
+    const body = new FormData();
+    body.append("user", "5d2ee9f3-7f36-4861-ad59-4297aa96f932");
+    body.append("contest", contestId);
+    body.append("content", createObjectURL);
+    body.append("description", submissionDescription);
+    postSubmission(body).then(() => Router.reload(window.location.pathname));
+  }
+
+  return (
+    <div className={styles.container}>
+      <Head>
+        <title>Revent</title>
+      </Head>
+
+      <main className={styles.main}>
+        <h1 className={styles.title}>Welcome to Revent!</h1>
+        <div className="grid grid-cols-3 gap-4">
+          {submissions.map(({ id, user, content, description }) => (
+            <div className="w-96 h-96 relative">
+              <BlurImage
+                key={id}
+                contestId={contestId}
+                submissionId={id}
+                user={getUserName(user, userList)}
+                url={content}
+                description={description}
+              />
+            </div>
+          ))}
+        </div>
+        <div>
+          <h4>Select Image</h4>
+          <input type="file" name="myImage" onChange={uploadToClient} />
+
+          {createObjectURL && (
+            <div>
+              <p>Description:</p>
+              <input
+                type="text"
+                name="description"
+                className="w-96 border-2 border-b-black"
+                onChange={changeDescription}
+              />
+              <img className="w-96 h-96" src={createObjectURL} />
+              <button
+                className="btn btn-primary w-96 border-2 border-black"
+                type="submit"
+                onClick={() => uploadToServer({ id: "ID" })}
+              >
+                Send to server
+              </button>
+            </div>
+          )}
+        </div>
+      </main>
+      <footer className={styles.footer}></footer>
+    </div>
+  );
+}
+
+export async function getServerSideProps(context) {
+  let contestId = context.params.id;
+  let submissions;
+  let userList;
+  try {
+    submissions = await getSubmissionsFromContest(contestId);
+    userList = await getUserList();
+  } catch (err) {
+    console.log(err);
+  }
+  return {
+    props: {
+      contestId,
+      submissions,
+      userList,
+    },
+  };
+}
+
+function BlurImage({ contestId, submissionId, url, user, description }) {
```

Comment:
This seems like a Component, let's create a components folder and place all the components inside